staging/src/k8s.io/apiserver/pkg/authentication/authenticator/interfaces.go

```diff
@@ -77,4 +78,29 @@ type Response struct {
 	Audiences Audiences
 	// User is the UserInfo associated with the authentication context.
 	User user.Info
+	// AuthenticatorName identifies the authenticator that authenticated the request.
+	AuthenticatorName string
+}
+
+// Error is the error returned by an authenticator upon failures.
+type Error struct {
+	// Err is the error produced by an authenticator.
+	Err error
+	// AuthenticatorName identifies the authenticator that failed an authentication request.
+	AuthenticatorName string
```

Comment:
same split of authn method and specific authenticator name probably makes sense here

Reply:
done.